pkg/yang/file.go

```diff
@@ -115,22 +124,43 @@ func findFile(name string) (string, string, error) {
 	return "", "", fmt.Errorf("no such file: %s", name)
 }
 
-// findInDir looks for a file named name in dir or any of its subdirectories.
-func findInDir(dir, name string) string {
+// findInDir looks for a file named name in dir or any of its subdirectories if
+// recurse is true. if recurse is false, scan only the directory dir.
+func findInDir(dir, name string, recurse bool) string {
 	fis, err := ioutil.ReadDir(dir)
 	if err != nil {
 		return ""
 	}
+	var candidates []string
+	mname := name
+	if strings.HasSuffix(mname, ".yang") {
+		mname = name[:len(name)-len(".yang")]
+	}
+
 	for _, fi := range fis {
 		if !fi.IsDir() {
 			if fi.Name() == name {
 				return filepath.Join(dir, name)
+			} else if !strings.Contains(name, "@") && strings.HasPrefix(fi.Name(), mname+"@") && strings.HasSuffix(fi.Name(), ".yang") {
+				candidates = append(candidates, fi.Name())
 			}
 			continue
 		}
-		if n := findInDir(filepath.Join(dir, fi.Name()), name); n != "" {
-			return n
+		if recurse {
+			if n := findInDir(filepath.Join(dir, fi.Name()), name, recurse); n != "" {
+				return n
+			}
 		}
 	}
+	if len(candidates) == 0 {
+		return ""
+	}
+	// sort the candidates and return the ultimate ("highest") value, which in
+	// the case of revision-date fields per the module file name format (RFC6020
+	// section 5.2) should be the newest candidate.
+	sort.Strings(candidates)
+	if best := candidates[len(candidates)-1]; best != "" {
+		return filepath.Join(dir, best)
+	}
```

Comment:
Why would best ever be ""?  That could only happen if fi.Name() returns "", which would not match the .yang suffice or the name@ prefix.